accounts/store.go

```diff
@@ -0,0 +1,311 @@
+package accounts
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/binary"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"go.etcd.io/bbolt"
+)
+
+const (
+	// DBFilename is the filename within the data directory which contains
+	// the macaroon stores.
+	DBFilename = "accounts.db"
+
+	// dbFilePermission is the default permission the account database file
+	// is created with.
+	dbFilePermission = 0600
+
+	// dbPathPermission is the default permission the account database
+	// directory is created with (if it does not exist).
+	dbPathPermission = 0700
+
+	// DefaultAccountDBTimeout is the default maximum time we wait for the
+	// account bbolt database to be opened. If the database is already
+	// opened by another process, the unique lock cannot be obtained. With
+	// the timeout we error out after the given time instead of just
+	// blocking for forever.
+	DefaultAccountDBTimeout = 5 * time.Second
+)
+
+var (
+	// accountBucketName is the name of the bucket where all accounting
+	// based balances are stored.
+	accountBucketName = []byte("accounts")
+
+	// lastAddIndexKey is the name of the key under which we store the last
+	// known invoice add index.
+	lastAddIndexKey = []byte("last-add-index")
+
+	// lastSettleIndexKey is the name of the key under which we store the
+	// last known invoice settle index.
+	lastSettleIndexKey = []byte("last-settle-index")
+
+	// byteOrder is the binary byte order we use to encode integers.
+	byteOrder = binary.BigEndian
+)
+
+// BoltStore wraps the bolt DB that stores all accounts and their balances.
+type BoltStore struct {
+	db *bbolt.DB
```

Comment:
What if we modified this to use the `kvdb.DB` interface? We can then have an implementation that in practice will be bolt initially, but would let users hook this into their existing postgres (and later on `sqlite`) DB and inherit replication and some very basic querying (would at least be able to read the account values independently).